pkg/apis/pipeline/v1alpha1/task_validation.go

```diff
@@ -27,6 +27,8 @@ import (
 	"k8s.io/apimachinery/pkg/util/validation"
 )
 
+const substitution = "[_a-zA-Z][_a-zA-Z0-9.-]*"
```

Comment:
I think this is a good "base" for variable name but might be too restrictive though

Reply:
seems fine to me, it allows `-` and `_`